judoscale/celery/tasks_monitor.py

```diff
@@ -0,0 +1,87 @@
+import logging
+from datetime import datetime as dt
+#from celery import Celery
+from celery import current_app
+
+from judoscale.core.metric import Metric
+from judoscale.core.reporter import reporter
+from judoscale.core.metrics_store import metrics_store
+
+logger = logging.getLogger(__name__)
+
+
+class EventLogger:
+    def log_task_status_change(self, task, event):
+        print('[{}] {} {} (STATE={}, UUID={})'.format(
+            self._to_datetime(task.timestamp),
+            event['type'].upper(),
+            task.name,
+            task.state.upper(),
+            task.uuid
+        ))
+    def _to_datetime(self, timestamp):
+        return dt.fromtimestamp(timestamp) if timestamp is not None else None
+
+
+class CeleryEventsHandler:
+    def __init__(self, celery_app, verbose_logging=False):
+        self._app = celery_app
+        self._state = celery_app.events.State()
+        self._logger = EventLogger()
+        self._verbose_logging = verbose_logging
+        self.task_started = {}
+        self.task_created = {}
+
+    def announce_task_received(self, event):
+        self._state.event(event)
+        task = self._state.tasks.get(event["uuid"])
+        self.task_created[task.uuid] = task.timestamp
+        self._logger.log_task_status_change(task, event)
+        now = dt.now()
+        current_timestamp_ms = now.timestamp()
+        queue_time_ms = (
+            self.task_created[task.uuid] - current_timestamp_ms) * 1000
+        metric = Metric(
+            measurement="queue_time",
+            datetime=dt.fromtimestamp(self.task_created[task.uuid]),
+            value=queue_time_ms
+        )
+        logger.debug("queue_time={}ms".format(round(queue_time_ms, 2)))
+        if metric:
+            metrics_store.add_queue(metric)
+        reporter.ensure_running()
```

Comment:
@adamlogic metrics and reporter are added here.
The user needs to make a `monitor.py` file in the sample app and import the module `CeleryEventsHandler` from this file here (see the `monitor.py` below).